pkg/nuclide/file-tree/lib/FileTreeController.js

```diff
@@ -741,16 +747,62 @@ class FileTreeController {
       onClose: () => this._closeDialog(),
       shouldSelectBasename: true,
     };
-    this._openDialog(<FileDialogComponent {...props} />);
+    this._openDialog(props);
   }
 
-  _openDialog(component: ReactElement): void {
+  openDuplicateDialog(): void {
+    var selection = this._getSelectedEntryAndDirectoryAndRoot();
+    if (!selection || selection.entry.isDirectory()) {
+      return;
+    }
+
+    var message = (
+      <div>
+        <div>Enter the new name for the file in the root:</div>
+        <div>{path.normalize(selection.root.getPath() + '/')}</div>
+      </div>
+    );
+
+    var {entry, root} = selection;
+
+    var props = {
+      rootDirectory: root,
+      initialEntry: entry,
+      message,
+      onConfirm: (rootDirectory, relativeFilePath) => {
+        var file = rootDirectory.getFile(relativeFilePath);
+        file.create().then(async (createdSuccessfully) => {
```

Comment:
Ah, yeah. Didn't know `var createdSuccessfully = await file.create();` was valid syntax.. It's changed now. :sunglasses: